client/ConnectionThread.py

```diff
@@ -0,0 +1,109 @@
+            if not self.handler_status.is_connected:
+                if self.handler_status.is_active_connect:
+                    self.connect()
+                else:
+                    self.listen()
+            else:
+                break
+        self.conn.settimeout(3.0)
+        while not self.handler_status.to_kill:
+            self.send_counter()
+            time.sleep(1.0)
+            self.get_counter()
+            self.handler_status.msg += 1
+        self.kill()
+
+    def send_counter(self):
+        self.lock.acquire()
+        for i in range(5):
+            try:
+                msg = f"{self.handler_status.host_username} -> {self.handler_status.player_name}:{self.handler_status.msg}"
+                sent = self.conn.send(msg.encode())
+                print(f"Sent: {sent}")
+                break
+            except socket.error:
+                if i == 5:
+                    print("Couldn't send message. Receiver must've closed connection!")
+                    self.handler_status.is_connected = -1
+        self.lock.release()
+
+    def get_counter(self):
+        self.lock.acquire()
+        for i in range(5):
+            try:
+                msg = self.conn.recv(2048)
+                msg = msg.decode()
+                if msg == '' or msg == ' ':
+                    print("Socket connection shut down")
+                    self.handler_status.is_connected = -1
+                print(msg)
+                break
+            except socket.error:
+                if i == 5:
+                    print("Couldn't receive message. Receiver must've closed connection!")
+                    self.handler_status.is_connected = -1
+        self.lock.release()
+
+    def connect(self):
+        tries = 0
+        self.socket.settimeout(10.0)
+        print(f"Trying to connect with: {self.handler_status.player_ip}:{self.handler_status.player_port}")
+        while True:
+            try:
+                self.socket.connect((self.handler_status.player_ip, self.handler_status.player_port))
+                self.socket.settimeout(None)
+                self.conn = self.socket
+                print(f"Connected with: {self.handler_status.player_ip}:{self.handler_status.player_port}")
+                self.handler_status.is_connected = 1
+                return
+            except socket.error as msg:
+                print(f"Try: {tries}")
+                print(f"Couldn't connect: {msg}")
+                tries += 1
+                if tries > 5:
+                    self.socket.close()
+                    self.handler_status.is_connected = -1
+                    self.handler_status.to_kill = 1
+                    return
+                time.sleep(1.0)
+
+    def listen(self):
+        self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.socket.bind((self.handler_status.host_ip, self.handler_status.host_port))
+        self.socket.settimeout(10.0)
+        self.socket.listen(1)
```

Comment:
consider changing value for bigger in case of random connection errors

Reply:
Timeout time is good, backlog of listening changed to a bigger number, but I'm not sure that 1 socket can try to establish 2 connections at the same time though.